src/core/org/apache/jmeter/gui/action/CheckDirty.java

```diff
@@ -64,11 +64,12 @@
     public CheckDirty() {
         previousGuiItems = new HashMap<>();
         ActionRouter.getInstance().addPreActionListener(ExitCommand.class, this);
+        ActionRouter.getInstance().addPostActionListener(UndoCommand.class, this);
     }
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        if (e.getActionCommand().equals(ActionNames.EXIT)) {
+        if (e.getActionCommand().equals(ActionNames.EXIT) || e.getActionCommand().equals(ActionNames.UNDO) || e.getActionCommand().equals(ActionNames.REDO)) {
```

Comment:
Maybe we should define one set of names that identify the commands that should call `doAction` than this would be probably more readable.